tests/task_plugin_test.py

```diff
@@ -1,138 +1,145 @@
-import functools
-import mock
-import unittest
+import re
 
-from girder_worker_utils import decorators
-from girder_worker_utils import types
 from girder_worker import entrypoint
 from girder_worker.__main__ import main
 from girder_worker.app import app
-from six import StringIO
-
-
-class set_namespace(object):
-    def __init__(self, namespace):
-        self.namespace = namespace
-
-    def __call__(self, func):
-        @functools.wraps(func)
-        def wrapped(*args, **kwargs):
-            original = entrypoint.NAMESPACE
-            entrypoint.NAMESPACE = self.namespace
-            try:
-                result = func(*args, **kwargs)
-            finally:
-                entrypoint.NAMESPACE = original
-            return result
-        return wrapped
-
-
-class TestTaskPlugin(unittest.TestCase):
-    @set_namespace('girder_worker._test_plugins.valid_plugins')
-    def test_get_extension_manager(self):
-        mgr = entrypoint.get_extension_manager()
-        names = sorted(mgr.names())
-        self.assertEqual(names, ['core', 'plugin1', 'plugin2'])
-
-    @set_namespace('girder_worker._test_plugins.valid_plugins')
-    def test_get_core_task_modules(self):
-        modules = entrypoint.get_core_task_modules()
-        self.assertEqual(modules, ['os.path'])
-
-    @set_namespace('girder_worker._test_plugins.valid_plugins')
-    @mock.patch('girder_worker.entrypoint.import_module')
-    def test_import_all_includes(self, imp):
+
+from girder_worker_utils import decorators
+from girder_worker_utils import types
+
+import mock
+import pytest
+import six
+
+
+def setup_function(func):
+    if hasattr(func, 'namespace'):
+        namespace = func.namespace.args[0]
+        func.original = entrypoint.NAMESPACE
+        entrypoint.NAMESPACE = namespace
```

Comment:
This doesn't appear to be used anywhere.

Reply:
Oh, this is probably a pytest thing...  Ignore me.

Reply:
This is run before each test automatically by pytest.  It checks for the namespace mark on the test function and then sets the namespace to that.  ```teardown_function``` cleans it up like a context manager